### PR TITLE
Developer based documentation

### PR DESCRIPTION
**Pull request by 91pavan**
- Adding documentation for developers working with SMH APIs.
- Will be updated going forwards as we hope it will be useful for others.

Feel free to move it elsewhere in the repo where you see fit.

### CODE REVIEW

File: DEVELOPERS.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Working with SMH APIs
+
+If you want to work with SMH created objects like `workloads`, `traffictargets` etc, you can use the SMH APIs to do so.
```
Comment (joekelley) [inline DEVELOPERS.md:3]:
```suggestion
If you want to write a custom Kubernetes controller for SMH created objects like `workloads`, `traffictargets` etc, you can use the SMH APIs to do so.
```

Comment (joekelley) [reply DEVELOPERS.md:3]: It would also be great to highlight in this section that we welcome PRs for extending the built-in SMH discovery controller

**Review by joekelley — COMMENTED** [review]
Thanks for this contribution @91pavan!

Since this doc is focused on the use case of extending SMH's discovery mechanism with a custom discovery controller, let's move it to `{repo root}/dev-docs/extending-discovery.md`.

**Comment by 91pavan** [comment]
Thanks @joekelley . I've updated the document and moved it to `dev-docs` folder.

**Review by joekelley — APPROVED** [review]

**Pull request merged**